src/core/Algorithm.tsx

```diff
@@ -1,19 +1,48 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 
 import { RepresentationContext, DownstreamContext } from './View';
 import { smartEqualsShallow } from '../utils';
 
 import vtk from '@kitware/vtk.js/vtk.js';
 
+interface AlgorithmProps {
+  /**
+   * The ID used to identify this component.
+   */
+  id?: string;
+  /**
+   * downstream connection port
+   */
+  port?: number;
+  /**
+   * vtkClass name
+   */
+  vtkClass?: string;
+  /**
+   * set of property values for vtkClass
+   */
+  state?: object;
```

Comment:
I used a codemod to convert prop types to TS types, they will need some improvemements.